cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/action/ActionContext.java

```diff
@@ -35,4 +42,31 @@
   @Override
   SettableArguments getArguments();
 
+  /**
+   * This method is based on the assumption that the plugin has an input schema or output schema and the DAG of the
+   * operations are known. Actions do not have input and output schema, and the DAG is unknown to us. Therefore,
+   * this method cannot be used to emit field lineage information.
+   *
+   * @param fieldOperations the operations to be recorded
+   * @deprecated use {@link #record(Collection)} to emit field operations for actions, that method requires the input
+   *             field to contain the previous originated operation name. Using that DAG can be computed so the field
+   *             linage can be computed.
+   */
+  @Override
+  @Deprecated
+  default void record(List<FieldOperation> fieldOperations) {
+    throw new UnsupportedOperationException("Lineage recording is not supported.");
+  }
+
+  /**
+   * Register lineage for this Spark program using the given reference name
+   *
+   * @param referenceName reference name used for source
+   *
+   * @param accessType the access type of the lineage
+   * @throws DatasetManagementException thrown if there was an error in creating reference dataset
+   * @throws TransactionFailureException thrown if there was an error while fetching the dataset to register usage
+   */
+  void registerLineage(String referenceName,
```

Comment:
Better to use a default whenever adding a new method to an interface, to prevent compilation failures in downstream projects. Can you add a default no-op implementation here as well that throws an exception